### ISSUE TITLE
Non-aliased imported symbol not considered private

### ISSUE THREAD

**Issue by tibdex** [microsoft/pyright#2277]
**Describe the bug**

Pyright doesn't flag a non-aliased imported symbol as private.

**To Reproduce**

```python
# sub.py (lib)
ONE = 1
TWO = 2
```

```python
# __init__.py (lib)
# ONE is re-exported, TWO is not.
from .sub import ONE as ONE, TWO

THREE = ONE + TWO
```

Where lib has a `py.typed` file as described in [PEP 561](https://www.python.org/dev/peps/pep-0561/).

```python
# __main__.py (app)
import lib

print(lib.ONE, lib.TWO, lib.THREE)
```

**Expected behavior**

According to:

https://github.com/microsoft/pyright/blob/cecb3e1e0cfd6dd504655ba678332f3399e218dc/docs/typed-libraries.md#L35

and [this section of PEP 484](https://www.python.org/dev/peps/pep-0484/#stub-files):

> Modules and variables imported into the stub are not considered exported from the stub unless the import uses the `import ... as ...` form or the equivalent `from ... import ... as ...` form. (_UPDATE_: To clarify, the intention here is that only names imported using the form `X as X` will be exported, i.e. the name before and after as must be the same.)

`reportPrivateUsage` (or another rule) should say that `lib.TWO` is private.

**VS Code extension or command-line**

Command line with this config:

```toml
[tool.pyright]
typeCheckingMode = "strict"
useLibraryCodeForTypes = true
```

version: 1.1.166.

**Additional context**

mypy 0.910 correctly reports the issue:

```
app/__main__.py:3: error: Module has no attribute "TWO"
```

**Comment by erictraut** [comment]
Thanks for the bug report. You're correct that there was a missing check. I've implemented a fix. We're going to need to take care in rolling out this bug fix because it's likely to generate new errors within some code bases.

@jakebailey, here's the PR: https://github.com/microsoft/pyright/pull/2279. I plan to wait until after this week's pylance release before merging this.

**Comment by erictraut** [comment]
We've decided to add a new diagnostic rule. It's called "reportPrivateImportUsage". It will be on by default with basic type checking rules, but it can be disabled to retain the existing behavior. This will give you the ability to work around cases where the maintainers of a "py.typed" library haven't properly added redundant alias forms to indicate which imported symbols should be re-exported.

**Comment by erictraut** [comment]
This is included in pyright 1.1.168, which I just published. It will also be included in the next release of pylance.

**Comment by GBeauregard** [comment]
Is this a correct interpretation of the PEP? It seems to be in the section for notes on stub files, and it's not clear to me why this would should apply to libraries typed inline as well. At least some maintainers appear to believe it's a bug and the discussion therein wasn't clarified despite crosslinking here https://github.com/willmcgugan/rich/issues/1523

**Comment by GBeauregard** [comment]
Thanks, this clarifies things a lot. I do sympathize with the linked maintainer, though. The symbols there are only imported as part of an intro message program in `__main__.py`. Is it worth blacklisting `__main__.py` files from this check? For what it's worth `mypy --strict` doesn't throw any errors with that particular library so they may be doing something similar.
